cobalt/common/libc/stdlib/environment.cc

```diff
@@ -0,0 +1,203 @@
+// Copyright 2025 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <unistd.h>
+
+#include <algorithm>
+#include <cerrno>
+#include <cstring>
+#include <string>
+#include <utility>
+#include <vector>
+
+// This file can not depend on //base since that indirectly depends on this
+// file, but since base::NoDestructor is a header-only class, we can safely use
+// it here.
+#include "base/no_destructor.h"  // nogncheck
+
+namespace cobalt {
+namespace common {
+namespace libc {
+namespace stdlib {
+
+namespace {
+
+class Environment {
+ public:
+  typedef std::vector<std::string> Strings;
+
+  static char* getenv(const char* name);
+  static int setenv(const char* name, const char* value, int overwrite);
+  static int unsetenv(const char* name);
+  static char** InitializeGlobalEnviron();
+
+ private:
+  static base::NoDestructor<Strings> stored_environment_;
+  static base::NoDestructor<std::vector<char*>> environ_pointers_;
```

Comment:
I doubt it'll make much difference because I assume we rarely use the environment, but is there any reason why you use vectors over maps for storage in this class?